mod/item.php

```diff
@@ -1015,11 +1015,7 @@ function handle_tag(App $a, &$body, &$inform, &$str_tags, $profile_uid, $tag, $n
 
 			$profile = $contact["url"];
 			$alias   = $contact["alias"];
-			$newname = $contact["nick"];
-
-			if (($newname == "") || !in_array($contact["network"], [Protocol::ACTIVITYPUB, Protocol::OSTATUS, Protocol::TWITTER, Protocol::STATUSNET])) {
-				$newname = $contact["name"];
-			}
+			$newname = defaults($contact, "name", $contact["nick"]);
```

Comment:
Why had you done this change? The previous code had assured that on AP, OStatus and Twitter the generated code will contain the nick name and not the name, since this is the way these networks do work.

Reply:
The problem with this code is that it was considering the contact network, not the post's target network. By leaving all mentions as `@[url=Profile URL]Display Name[/url]`, this lets the BBCode parser/Addon handle the mentions to match the target network, without having decisions made upstream based on the contact's network.

For example, this allows me to mentions several people from different networks in a post, and they will all be displayed correctly when sent to Twitter, thanks to the custom `twitter_handle_mentions()` function. This wasn't the case before where the display would depend on the contact's network, and generally be wrong for Twitter.

Reply:
Okay. This with the target network is right. We have to make this possibly receiver depending. I really would like to see some protocol specific solution before this is merged.

Reply:
`BBCode::convert` already handle @-mentions depending on the target network (with the `$simplehtml` parameter), what `handle_tags()` was doing is a pre-conversion, removing BBCodes before the `BBCode::convert` pass.

Reply:
Yuuuup.